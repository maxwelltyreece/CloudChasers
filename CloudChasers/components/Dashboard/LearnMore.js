// React Imports
import React from 'react';
import { View, Text } from 'react-native';

// Style Imports
import dashboardStyles from '../../styles/DashboardStyles';

const RecentMealLog = () => (
    <View style={dashboardStyles.learnMoreContainer}>
      <Text style={dashboardStyles.learnMoreTitle}>Learn More</Text>
    </View>
);

export default RecentMealLog;
