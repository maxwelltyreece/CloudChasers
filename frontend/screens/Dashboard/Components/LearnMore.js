import React, { useState } from 'react';
import {
	View, Text, Pressable, Modal, StyleSheet, ScrollView,
} from 'react-native';

const styles = StyleSheet.create({
	learnMoreContainer: {
		flex: 1,
		// backgroundColor: '#EC6641',
		backgroundColor: 'white',
		justifyContent: 'center',
		alignContent: 'center',
		borderRadius: 15,
		// marginLeft: 10,
		padding: 20,
		right: '1%',
		width: '98%',
		shadowColor: '#000',
		shadowOffset: { width: 0, height: 1 },
		shadowOpacity: 0.22,
		shadowRadius: 2.22,
		elevation: 3,
	},
	learnMoreTitle: {
		fontSize: 20,
		fontWeight: 'bold',
	},
	learnMoreButton: {
		backgroundColor: '#F0F0F0',
		padding: 10,
		borderRadius: 32,
	},
	scrollView: {
		borderRadius: 10,
	},
	modalContainer: {
		flex: 1,
		marginVertical: 75,
		marginHorizontal: 20,
		backgroundColor: 'white',
		borderRadius: 20,
		padding: 35,
		justifyContent: 'center',
		alignItems: 'center',
		alignContent: 'center',
		shadowColor: '#000',
		shadowOffset: {
			width: 0,
			height: 2,
		},
		shadowOpacity: 0.5,
		shadowRadius: 4,
		elevation: 5,
		borderTopEndRadius: 28,
	},
	modalHeaderText: {
		fontSize: 22,
		fontWeight: '700',
		marginBottom: 35,
		color: '#007BFF',
		alignSelf: 'center',
	},
	modalSubheaderText: {
		fontWeight: 'bold',
	},
	modalContentText: {
		fontSize: 16,
		marginBottom: 20,
		// textAlign: 'justify',
	},
	modalCloseButtonText: {
		color: '#FFFFFF',
		marginTop: 20,
		backgroundColor: '#007BFF',
		paddingVertical: 10,
		paddingHorizontal: 20,
		borderRadius: 20,
		overflow: 'hidden',
		alignSelf: 'center',
		textTransform: 'uppercase',
		fontWeight: 'bold',
		letterSpacing: 1,
		shadowColor: '#000',
		shadowOffset: {
			width: 0,
			height: 2,
		},
		shadowOpacity: 0.25,
		shadowRadius: 3.84,
		elevation: 5,
		opacity: 0.9,
	},
});

function LearnMore() {
	const [modalVisible, setModalVisible] = useState(false);

	return (
		<View style={styles.learnMoreContainer}>
			<Pressable onPress={() => setModalVisible(true)}>
				<Text style={styles.learnMoreTitle}>
    Learn More
					{'>'}
				</Text>
			</Pressable>

			<Modal
				animationType="slide"
				transparent
				visible={modalVisible}
				onRequestClose={() => setModalVisible(!modalVisible)}
			>

				<View style={styles.modalContainer}>
					<ScrollView showsVerticalScrollIndicator={false} style={styles.scrollView}>
						<Text style={styles.modalHeaderText}>Understanding Nutrition</Text>

						{/* Calories */}
						<Text style={styles.modalContentText}>
							<Text style={styles.modalSubheaderText}>Calories (kcal):</Text>
							{'\n'}
                            Calories are units of energy that measure how much energy food provides
                            to the body. They are essential for daily activities and bodily
                            functions.
						</Text>

						{/* Carbohydrates */}
						<Text style={styles.modalContentText}>
							<Text style={styles.modalSubheaderText}>Carbohydrates:</Text>
							{'\n'}
                            The body&apos;s main source of energy, broken down into glucose.
                            Found in fruits, vegetables, breads, and cereals.
						</Text>

						{/* Proteins */}
						<Text style={styles.modalContentText}>
							<Text style={styles.modalSubheaderText}>Proteins:</Text>
							{'\n'}
                            Essential for building tissues, making enzymes, and supporting immune
                            function. Good sources include meat, fish, dairy, legumes, and nuts.
						</Text>

						{/* Fats */}
						<Text style={styles.modalContentText}>
							<Text style={styles.modalSubheaderText}>Fats:</Text>
							{'\n'}
                            Important for cell growth, organ protection, and nutrient absorption.
                            Includes saturated and unsaturated fats, found in oils, butter, avocado,
                            and fish.
						</Text>

						{/* Fibre */}
						<Text style={styles.modalContentText}>
							<Text style={styles.modalSubheaderText}>Fibre:</Text>
							{'\n'}
                            A type of carbohydrate that the body can&apos;t digest. It helps
                            regulate the body&apos;s use of sugars, helping to keep hunger
                            and blood sugar in check.
						</Text>

						{/* Water */}
						<Text style={styles.modalContentText}>
							<Text style={styles.modalSubheaderText}>Water:</Text>
							{'\n'}
                            Crucial for life, water aids in digestion, absorption,
                            circulation, and excretion. It regulates body temperature and
                            maintains electrolyte balance.
						</Text>

						{/* Micronutrients */}
						<Text style={styles.modalContentText}>
							<Text style={styles.modalSubheaderText}>
								Micronutrients
								{'\n'}
								(Vitamins & Minerals):
							</Text>
							{'\n'}
                            Vital for health, development, and disease prevention. Vitamins
                            support immune function and energy production, while minerals
                            are important for growth, bone health, and fluid balance.
						</Text>

					</ScrollView>

					<Pressable onPress={() => setModalVisible(!modalVisible)}>
						<Text style={styles.modalCloseButtonText}>Close</Text>
					</Pressable>

				</View>

			</Modal>

		</View>
	);
}

export default LearnMore;
