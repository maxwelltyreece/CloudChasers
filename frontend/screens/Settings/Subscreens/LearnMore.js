import React from 'react';
import {
	View, Text, StyleSheet, ScrollView,
} from 'react-native';

const styles = StyleSheet.create({
	container: {
		justifyContent: 'center',
		alignItems: 'center',
		padding: 20,
	},
	learnMoreTitle: {
		fontFamily: 'Montserrat_700Bold',
		fontSize: 20,
		fontWeight: 'bold',
	},
	learnMoreButton: {
		backgroundColor: '#F0F0F0',
		padding: 10,
		borderRadius: 32,
	},
	scrollView: {
		borderRadius: 10,
		width: '100%',
	},
	headerText: {
		fontFamily: 'Montserrat_700Bold',
		fontSize: 20,
		fontWeight: '700',
		marginTop: 8,
		marginBottom: 12,
		color: '#FF815E',
		left: '4%',
		alignSelf: 'flex-start',
	},
	textContainer: {
		height: 'auto',
		width: '95%',
		alignSelf: 'center',
		marginBottom: 10,
		backgroundColor: 'white',
		padding: 18,
		borderRadius: 10,
		elevation: 5,
		shadowColor: '#000',
		shadowOffset: {
			width: 0,
			height: 2,
		},
		shadowOpacity: 0.25,
		shadowRadius: 3.84,
	},
	subheaderText: {
		fontFamily: 'Montserrat_700Bold',
		fontWeight: 'bold',
		marginBottom: 6,
	},
	contentText: {
		fontSize: 15,
	},
});

function LearnMore() {

	return (

		<View>
			<ScrollView style={styles.scrollView}>
				<Text style={styles.headerText}>Understanding Nutrition</Text>
				<Text style={styles.headerText}>Understanding Nutrition</Text>

				<View style={styles.textContainer}>
					{/* Calories */}
					<Text style={styles.subheaderText}>Calories (kcal):</Text>
					<Text style={styles.contentText}>
						<View style={styles.textContainer}>
							{/* Calories */}
							<Text style={styles.subheaderText}>Calories (kcal):</Text>
							<Text style={styles.contentText}>
					Calories are units of energy that measure how much energy food provides
					to the body. They are essential for daily activities and bodily
					functions.
							</Text>
						</View>
					</Text>
				</View>

				<View style={styles.textContainer}>
					{/* Carbohydrates */}
					<Text style={styles.subheaderText}>Carbohydrates (g):</Text>
					<Text style={styles.contentText}>
						<View style={styles.textContainer}>
							{/* Carbohydrates */}
							<Text style={styles.subheaderText}>Carbohydrates (g):</Text>
							<Text style={styles.contentText}>
					The body&apos;s main source of energy, broken down into glucose.
					Found in fruits, vegetables, breads, and cereals.
							</Text>
						</View>
					</Text>
				</View>

				<View style={styles.textContainer}>
					{/* Protein */}
					<Text style={styles.subheaderText}>Protein (g):</Text>
					<Text style={styles.contentText}>
					Essential for building tissues, making enzymes, and supporting immune
					function. Good sources include meat, fish, dairy, legumes, and nuts.
					</Text>
				</View>

				<View style={styles.textContainer}>
					{/* Fats */}
					<Text style={styles.subheaderText}>Fats (g):</Text>
					<Text style={styles.contentText}>
					Important for cell growth, organ protection, and nutrient absorption.
					Includes saturated and unsaturated fats, found in oils, butter, avocado,
					and fish.
					</Text>
				</View>

				<View style={styles.textContainer}>
					{/* Fibre */}
					<Text style={styles.subheaderText}>Fibre (g):</Text>
					<Text style={styles.contentText}>
					A type of carbohydrate that the body can&apos;t digest. It helps
					regulate the body&apos;s use of sugars, helping to keep hunger
					and blood sugar in check.
					</Text>
				</View>

				<View style={styles.textContainer}>
					{/* Water */}
					<Text style={styles.subheaderText}>Water (ml):</Text>
					<Text style={styles.contentText}>
					Crucial for life, water aids in digestion, absorption,
					circulation, and excretion. It regulates body temperature and
					maintains electrolyte balance.
					</Text>
				</View>

				<View style={styles.textContainer}>
					{/* Sugars */}
					<Text style={styles.subheaderText}>Sugars (g):</Text>
					<Text style={styles.contentText}>
					Simple carbohydrates that provide a quick source of energy.
					Found in fruits, vegetables, and dairy products.
					</Text>
				</View>

				<View style={styles.textContainer}>
					{/* Sodium */}
					<Text style={styles.subheaderText}>Sodium (mg):</Text>
					<Text style={styles.contentText}>
					An essential mineral that helps maintain the body&apos;s fluid balance.
					Found in salt, processed foods, and condiments.
					</Text>
				</View>

				<View style={styles.textContainer}>
					{/* Micronutrients */}
					<Text style={styles.subheaderText}>
						Micronutrients
						{'\n'}
						(Vitamins & Minerals):
					</Text>
					<Text style={styles.contentText}>
					Vital for health, development, and disease prevention. Vitamins
					support immune function and energy production, while minerals
					are important for growth, bone health, and fluid balance.
					</Text>
				</View>

			</ScrollView>

		</View>
	);
}

export default LearnMore;
