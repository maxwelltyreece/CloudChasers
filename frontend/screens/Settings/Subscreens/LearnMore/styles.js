import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        justifyContent: 'center',
        alignItems: 'center',
        padding: 20,
    },
    learnMoreTitle: {
        fontFamily: 'Montserrat_700Bold',
        fontSize: 20,
        fontWeight: 'bold',
    },
    learnMoreButton: {
        backgroundColor: '#F0F0F0',
        padding: 10,
        borderRadius: 32,
    },
    scrollView: {
        borderRadius: 10,
        width: '100%',
    },
    headerText: {
        fontFamily: 'Montserrat_700Bold',
        fontSize: 20,
        fontWeight: '700',
        marginTop: 8,
        marginBottom: 12,
        color: '#FF815E',
        left: '4%',
        alignSelf: 'flex-start',
    },
    textContainer: {
        height: 'auto',
        width: '95%',
        alignSelf: 'center',
        marginBottom: 10,
        backgroundColor: 'white',
        padding: 18,
        borderRadius: 10,
        elevation: 5,
        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
    },
    subheaderText: {
        fontFamily: 'Montserrat_700Bold',
        fontWeight: 'bold',
        marginBottom: 6,
    },
    contentText: {
        fontSize: 15,
    },
});
