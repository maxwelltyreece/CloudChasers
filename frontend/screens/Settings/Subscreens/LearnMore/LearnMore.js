import React from 'react';
import { View, Text, ScrollView,} from 'react-native';
import { styles } from './styles';

/**
 * LearnMore component
 * @returns {JSX.Element} The LearnMore component
 */
function LearnMore() {
	return (
		<View>
			<ScrollView style={styles.scrollView}>
				<Text style={styles.headerText}>Understanding Nutrition</Text>

				<View style={styles.textContainer}>
					<Text style={styles.subheaderText}>Calories (kcal):</Text>
					<Text style={styles.contentText}>
						Calories are units of energy that measure how much energy food provides
						to the body. They are essential for daily activities and bodily
						functions.
					</Text>
				</View>

				<View style={styles.textContainer}>
					<Text style={styles.subheaderText}>Carbohydrates (g):</Text>
					<Text style={styles.contentText}>
						The body&apos;s main source of energy, broken down into glucose.
						Found in fruits, vegetables, breads, and cereals.
					</Text>
				</View>

				<View style={styles.textContainer}>
					<Text style={styles.subheaderText}>Protein (g):</Text>
					<Text style={styles.contentText}>
						Essential for building tissues, making enzymes, and supporting immune
						function. Good sources include meat, fish, dairy, legumes, and nuts.
					</Text>
				</View>

				<View style={styles.textContainer}>
					<Text style={styles.subheaderText}>Fats (g):</Text>
					<Text style={styles.contentText}>
						Important for cell growth, organ protection, and nutrient absorption.
						Includes saturated and unsaturated fats, found in oils, butter, avocado,
						and fish.
					</Text>
				</View>

				<View style={styles.textContainer}>
					<Text style={styles.subheaderText}>Fibre (g):</Text>
					<Text style={styles.contentText}>
						A type of carbohydrate that the body can&apos;t digest. It helps
						regulate the body&apos;s use of sugars, helping to keep hunger
						and blood sugar in check.
					</Text>
				</View>

				<View style={styles.textContainer}>
					<Text style={styles.subheaderText}>Water (ml):</Text>
					<Text style={styles.contentText}>
						Crucial for life, water aids in digestion, absorption,
						circulation, and excretion. It regulates body temperature and
						maintains electrolyte balance.
					</Text>
				</View>

				<View style={styles.textContainer}>
					<Text style={styles.subheaderText}>Sugars (g):</Text>
					<Text style={styles.contentText}>
						Simple carbohydrates that provide a quick source of energy.
						Found in fruits, vegetables, and dairy products.
					</Text>
				</View>

				<View style={styles.textContainer}>
					<Text style={styles.subheaderText}>Sodium (mg):</Text>
					<Text style={styles.contentText}>
						An essential mineral that helps maintain the body&apos;s fluid balance.
						Found in salt, processed foods, and condiments.
					</Text>
				</View>

				<View style={styles.textContainer}>
					<Text style={styles.subheaderText}>
						Micronutrients
						{'\n'}
						(Vitamins & Minerals):
					</Text>
					<Text style={styles.contentText}>
						Vital for health, development, and disease prevention. Vitamins
						support immune function and energy production, while minerals
						are important for growth, bone health, and fluid balance.
					</Text>
				</View>
			</ScrollView>
		</View>
	);
}

export default LearnMore;
