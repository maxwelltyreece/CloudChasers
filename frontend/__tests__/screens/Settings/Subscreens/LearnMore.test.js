/* eslint-disable no-undef */
import React from 'react';
import { render, act, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import LearnMore from '../../../../screens/Settings/Subscreens/LearnMore';


// Mock navigation
const Stack = createStackNavigator();

// Mock the useNavigation hook
jest.mock('@react-navigation/native', () => {

	const navigate = jest.fn();

	return {
		...jest.requireActual('@react-navigation/native'),
		useNavigation: () => ({
			navigate,
		}),
	};
});

// Mock AsyncStorage
jest.mock('@react-native-async-storage/async-storage', () => ({
	getItem: jest.fn(() => Promise.resolve('token')), // Simulate a token present
}));

describe('About Subscreen', () => {

	it('renders the About page correctly', async () => {
		const { getByText } = render(
			<NavigationContainer>
				<Stack.Navigator>
					<Stack.Screen name="LearnMore" component={LearnMore} />
				</Stack.Navigator>
			</NavigationContainer>
		);


		await act(async () => {
			await waitFor(() => { expect(getByText('Understanding Nutrition')).toBeTruthy(); });
			await waitFor(() => { expect(getByText('Calories (kcal):')).toBeTruthy(); });
			await waitFor(() => { expect(getByText('Protein (g):')).toBeTruthy(); });
			await waitFor(() => { expect(getByText('Carbohydrates (g):')).toBeTruthy(); });
			await waitFor(() => { expect(getByText('Fats (g):')).toBeTruthy(); });
			await waitFor(() => { expect(getByText('Fibre (g):')).toBeTruthy(); });
			await waitFor(() => { expect(getByText('Sugars (g):')).toBeTruthy(); });
			await waitFor(() => { expect(getByText('Sodium (mg):')).toBeTruthy(); });
			await waitFor(() => { expect(getByText('Water (ml):')).toBeTruthy(); });
			await waitFor(() => { expect(getByText('Micronutrients (Vitamins & Minerals):')).toBeTruthy(); });
		});
	});

	it('matches the about snapshot', async () => {
		const Stack = createStackNavigator();
		const tree = render(
			<NavigationContainer>
				<Stack.Navigator>
					<Stack.Screen name="LearnMore" component={LearnMore} />
				</Stack.Navigator>
			</NavigationContainer>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

});
