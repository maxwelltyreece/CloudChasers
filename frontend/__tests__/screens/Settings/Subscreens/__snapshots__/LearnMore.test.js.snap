// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Learn More Subscreen matches the about snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  LearnMore
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            collapsable={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              [
                {
                  "display": "flex",
                  "overflow": undefined,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              {
                "close": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                handlerTag={2}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  {
                    "flex": 1,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    [
                      {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      [
                        {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <View>
                        <RCTScrollView
                          style={
                            {
                              "borderRadius": 10,
                              "width": "100%",
                            }
                          }
                        >
                          <View>
                            <Text
                              style={
                                {
                                  "alignSelf": "flex-start",
                                  "color": "#FF815E",
                                  "fontFamily": "Montserrat_700Bold",
                                  "fontSize": 20,
                                  "fontWeight": "700",
                                  "left": "4%",
                                  "marginBottom": 12,
                                  "marginTop": 8,
                                }
                              }
                            >
                              Understanding Nutrition
                            </Text>
                            <View
                              style={
                                {
                                  "alignSelf": "center",
                                  "backgroundColor": "white",
                                  "borderRadius": 10,
                                  "elevation": 5,
                                  "height": "auto",
                                  "marginBottom": 10,
                                  "padding": 18,
                                  "shadowColor": "#000",
                                  "shadowOffset": {
                                    "height": 2,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 0.25,
                                  "shadowRadius": 3.84,
                                  "width": "95%",
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "fontFamily": "Montserrat_700Bold",
                                    "fontWeight": "bold",
                                    "marginBottom": 6,
                                  }
                                }
                              >
                                Calories (kcal):
                              </Text>
                              <Text
                                style={
                                  {
                                    "fontSize": 15,
                                  }
                                }
                              >
                                Calories are units of energy that measure how much energy food provides to the body. They are essential for daily activities and bodily functions.
                              </Text>
                            </View>
                            <View
                              style={
                                {
                                  "alignSelf": "center",
                                  "backgroundColor": "white",
                                  "borderRadius": 10,
                                  "elevation": 5,
                                  "height": "auto",
                                  "marginBottom": 10,
                                  "padding": 18,
                                  "shadowColor": "#000",
                                  "shadowOffset": {
                                    "height": 2,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 0.25,
                                  "shadowRadius": 3.84,
                                  "width": "95%",
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "fontFamily": "Montserrat_700Bold",
                                    "fontWeight": "bold",
                                    "marginBottom": 6,
                                  }
                                }
                              >
                                Carbohydrates (g):
                              </Text>
                              <Text
                                style={
                                  {
                                    "fontSize": 15,
                                  }
                                }
                              >
                                The body's main source of energy, broken down into glucose. Found in fruits, vegetables, breads, and cereals.
                              </Text>
                            </View>
                            <View
                              style={
                                {
                                  "alignSelf": "center",
                                  "backgroundColor": "white",
                                  "borderRadius": 10,
                                  "elevation": 5,
                                  "height": "auto",
                                  "marginBottom": 10,
                                  "padding": 18,
                                  "shadowColor": "#000",
                                  "shadowOffset": {
                                    "height": 2,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 0.25,
                                  "shadowRadius": 3.84,
                                  "width": "95%",
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "fontFamily": "Montserrat_700Bold",
                                    "fontWeight": "bold",
                                    "marginBottom": 6,
                                  }
                                }
                              >
                                Protein (g):
                              </Text>
                              <Text
                                style={
                                  {
                                    "fontSize": 15,
                                  }
                                }
                              >
                                Essential for building tissues, making enzymes, and supporting immune function. Good sources include meat, fish, dairy, legumes, and nuts.
                              </Text>
                            </View>
                            <View
                              style={
                                {
                                  "alignSelf": "center",
                                  "backgroundColor": "white",
                                  "borderRadius": 10,
                                  "elevation": 5,
                                  "height": "auto",
                                  "marginBottom": 10,
                                  "padding": 18,
                                  "shadowColor": "#000",
                                  "shadowOffset": {
                                    "height": 2,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 0.25,
                                  "shadowRadius": 3.84,
                                  "width": "95%",
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "fontFamily": "Montserrat_700Bold",
                                    "fontWeight": "bold",
                                    "marginBottom": 6,
                                  }
                                }
                              >
                                Fats (g):
                              </Text>
                              <Text
                                style={
                                  {
                                    "fontSize": 15,
                                  }
                                }
                              >
                                Important for cell growth, organ protection, and nutrient absorption. Includes saturated and unsaturated fats, found in oils, butter, avocado, and fish.
                              </Text>
                            </View>
                            <View
                              style={
                                {
                                  "alignSelf": "center",
                                  "backgroundColor": "white",
                                  "borderRadius": 10,
                                  "elevation": 5,
                                  "height": "auto",
                                  "marginBottom": 10,
                                  "padding": 18,
                                  "shadowColor": "#000",
                                  "shadowOffset": {
                                    "height": 2,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 0.25,
                                  "shadowRadius": 3.84,
                                  "width": "95%",
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "fontFamily": "Montserrat_700Bold",
                                    "fontWeight": "bold",
                                    "marginBottom": 6,
                                  }
                                }
                              >
                                Fibre (g):
                              </Text>
                              <Text
                                style={
                                  {
                                    "fontSize": 15,
                                  }
                                }
                              >
                                A type of carbohydrate that the body can't digest. It helps regulate the body's use of sugars, helping to keep hunger and blood sugar in check.
                              </Text>
                            </View>
                            <View
                              style={
                                {
                                  "alignSelf": "center",
                                  "backgroundColor": "white",
                                  "borderRadius": 10,
                                  "elevation": 5,
                                  "height": "auto",
                                  "marginBottom": 10,
                                  "padding": 18,
                                  "shadowColor": "#000",
                                  "shadowOffset": {
                                    "height": 2,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 0.25,
                                  "shadowRadius": 3.84,
                                  "width": "95%",
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "fontFamily": "Montserrat_700Bold",
                                    "fontWeight": "bold",
                                    "marginBottom": 6,
                                  }
                                }
                              >
                                Water (ml):
                              </Text>
                              <Text
                                style={
                                  {
                                    "fontSize": 15,
                                  }
                                }
                              >
                                Crucial for life, water aids in digestion, absorption, circulation, and excretion. It regulates body temperature and maintains electrolyte balance.
                              </Text>
                            </View>
                            <View
                              style={
                                {
                                  "alignSelf": "center",
                                  "backgroundColor": "white",
                                  "borderRadius": 10,
                                  "elevation": 5,
                                  "height": "auto",
                                  "marginBottom": 10,
                                  "padding": 18,
                                  "shadowColor": "#000",
                                  "shadowOffset": {
                                    "height": 2,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 0.25,
                                  "shadowRadius": 3.84,
                                  "width": "95%",
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "fontFamily": "Montserrat_700Bold",
                                    "fontWeight": "bold",
                                    "marginBottom": 6,
                                  }
                                }
                              >
                                Sugars (g):
                              </Text>
                              <Text
                                style={
                                  {
                                    "fontSize": 15,
                                  }
                                }
                              >
                                Simple carbohydrates that provide a quick source of energy. Found in fruits, vegetables, and dairy products.
                              </Text>
                            </View>
                            <View
                              style={
                                {
                                  "alignSelf": "center",
                                  "backgroundColor": "white",
                                  "borderRadius": 10,
                                  "elevation": 5,
                                  "height": "auto",
                                  "marginBottom": 10,
                                  "padding": 18,
                                  "shadowColor": "#000",
                                  "shadowOffset": {
                                    "height": 2,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 0.25,
                                  "shadowRadius": 3.84,
                                  "width": "95%",
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "fontFamily": "Montserrat_700Bold",
                                    "fontWeight": "bold",
                                    "marginBottom": 6,
                                  }
                                }
                              >
                                Sodium (mg):
                              </Text>
                              <Text
                                style={
                                  {
                                    "fontSize": 15,
                                  }
                                }
                              >
                                An essential mineral that helps maintain the body's fluid balance. Found in salt, processed foods, and condiments.
                              </Text>
                            </View>
                            <View
                              style={
                                {
                                  "alignSelf": "center",
                                  "backgroundColor": "white",
                                  "borderRadius": 10,
                                  "elevation": 5,
                                  "height": "auto",
                                  "marginBottom": 10,
                                  "padding": 18,
                                  "shadowColor": "#000",
                                  "shadowOffset": {
                                    "height": 2,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 0.25,
                                  "shadowRadius": 3.84,
                                  "width": "95%",
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "fontFamily": "Montserrat_700Bold",
                                    "fontWeight": "bold",
                                    "marginBottom": 6,
                                  }
                                }
                              >
                                Micronutrients
                                

                                (Vitamins & Minerals):
                              </Text>
                              <Text
                                style={
                                  {
                                    "fontSize": 15,
                                  }
                                }
                              >
                                Vital for health, development, and disease prevention. Vitamins support immune function and energy production, while minerals are important for growth, bone health, and fluid balance.
                              </Text>
                            </View>
                          </View>
                        </RCTScrollView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
